### PR TITLE
Alerting: Deprecate max_annotations_to_keep and max_annotation_age in [alerting] configuration section

### PR DESCRIPTION
**Pull request by yuri-tceretian**
This PR introduces a new section `[unified_alerting.state_history.annotations]` in the Grafana configuration file that will replace the settings in the legacy alerting section `max_annotation_age` and `max_annotations_to_keep`.

The code is updated to read the new section and if both fields are empty, i.e. default values, fall back to old settings. If the old settings are not default, it will emit a warning message into the log output:
```
"settings 'max_annotations_to_keep' and 'max_annotation_age' in section [alerting] are deprecated. Please use settings 'max_annotations_to_keep' and 'max_age' in section [unified_alerting.state_history.annotations]
```

I chose to rename field `max_annotation_age` to `max_age` to match similar settings in other places 

Contributes to https://github.com/grafana/alerting-squad/issues/707

### CODE REVIEW

**Review by JacobsonMT — APPROVED** [review]
LGTM, let's wait for copy review from docs before merging though.

**Review by rwwiv — APPROVED** [review]
LGTM!

File: conf/defaults.ini
Comment (brendamuir) [inline conf/defaults.ini:1306]:
```suggestion
# Configures how long alert annotations are stored for. Default is 0, which keeps them forever.
```

File: conf/defaults.ini
Comment (brendamuir) [inline conf/defaults.ini:1303]:
```suggestion
# Controls retention of annotations automatically created while evaluating alert rules
```

File: conf/defaults.ini
Comment (brendamuir) [inline conf/defaults.ini:1304]: not sure if we need the bracket here (a setting ?

**Review by brendamuir — APPROVED** [review]

File: conf/defaults.ini
Comment (brendamuir) [inline conf/defaults.ini:1307]:
```suggestion
# This setting should be expressed as a duration. Ex 6h (hours), 10d (days), 2w (weeks), 1M (month).
```

File: conf/defaults.ini
Comment (brendamuir) [inline conf/defaults.ini:1304]:
```suggestion
# Controls retention of annotations automatically created while evaluating alert rules. 
# Alert state history backend must be configured to be annotations (see setting [unified_alerting.state_history].backend).
```

Comment (brendamuir) [reply conf/defaults.ini:1304]: "must be configured to be annotations" sounds odd - what do you mean? Annotations must be configured in the alert state history backend? The alert state history backend must have annotations configured.?

Comment (yuri-tceretian) [reply conf/defaults.ini:1304]: alert state history has two backend modes: loki and annotations API. The new section only takes effect when backend is "annotations"

File: conf/defaults.ini
Comment (yuri-tceretian) [reply conf/defaults.ini]: I amended the text. Please check again.

**Comment by yuri-tceretian** [comment]
Ooops.. merge did not go as I expected. Sorry!

File: pkg/setting/setting.go
Comment (yuri-tceretian) [reply pkg/setting/setting.go:720]: Done

**Pull request merged**
